--- a/meson.build
+++ b/meson.build
@@ -140,7 +140,7 @@
 endif
 
 if os_qnx
-  wayland_enabled = false
+  #wayland_enabled = false
   vulkan_enabled = false
   x11_enabled = false
 else
--- a/gdk/wayland/gdkdisplay-wayland.c
+++ b/gdk/wayland/gdkdisplay-wayland.c
@@ -31,7 +31,9 @@
 #endif
 
 #include <sys/mman.h>
+#ifdef __linux__
 #include <sys/syscall.h>
+#endif
 
 #include <glib.h>
 #include <gio/gio.h>
--- a/gdk/wayland/gdkseat-wayland.c
+++ b/gdk/wayland/gdkseat-wayland.c
@@ -18,6 +18,7 @@
 #include "config.h"
 
 #include "gdkdevice-wayland-private.h"
+#include <linux/input-event-codes.h>
 
 #include "gdkclipboard-wayland.h"
 #include "gdkclipboardprivate.h"
